src/pages/settings/Payments/ChooseTransferAccountPage.js

```diff
@@ -1,27 +1,104 @@
 
 const propTypes = {
+    /** Wallet transfer propTypes */
+    walletTransfer: walletTransferPropTypes,
+
+    /** Are we loading payment methods? */
+    isLoadingPaymentMethods: PropTypes.bool,
+
     ...withLocalizePropTypes,
 };
 
-const ChooseTransferAccountPage = props => (
-    <ScreenWrapper>
-        <KeyboardAvoidingView>
-            <HeaderWithCloseButton
-                title={props.translate('chooseTransferAccountPage.chooseAccount')}
-                shouldShowBackButton
-                onBackButtonPress={() => Navigation.goBack()}
-                onCloseButtonPress={() => Navigation.dismissModal()}
-            />
-        </KeyboardAvoidingView>
-    </ScreenWrapper>
-);
+const defaultProps = {
+    walletTransfer: {},
+    isLoadingPaymentMethods: false,
+};
+
+const ChooseTransferAccountPage = (props) => {
+    /**
+     * Go back to TransferPage with the selected bank account
+     * @param {Object} event Click event object
+     * @param {String} accountID of the selected account.
+     */
+    const selectAccountAndNavigateBack = (event, accountID) => {
+        PaymentMethods.saveWalletTransferAccount(accountID);
+        Navigation.navigate(ROUTES.SETTINGS_PAYMENTS_TRANSFER_BALANCE);
+    };
+
+    /**
+     * @param {String} paymentType
+     */
+    const navigateToAddPaymentMethodPage = () => {
+        if (props.filterType === CONST.PAYMENT_METHODS.DEBIT_CARD) {
+            Navigation.navigate(ROUTES.SETTINGS_ADD_DEBIT_CARD);
+            return;
+        }
+        Navigation.navigate(ROUTES.SETTINGS_ADD_BANK_ACCOUNT);
+    };
+
+    return (
+        <ScreenWrapper>
+            <KeyboardAvoidingView>
+                <HeaderWithCloseButton
+                    title={props.translate('chooseTransferAccountPage.chooseAccount')}
+                    shouldShowBackButton
+                    onBackButtonPress={() => Navigation.goBack()}
+                    onCloseButtonPress={() => Navigation.dismissModal()}
+                />
+                <View style={[styles.mt6, styles.flexShrink1, styles.flexBasisAuto]}>
+                    <PaymentMethodList
+                        onPress={selectAccountAndNavigateBack}
+                        shouldShowSelectedState
+                        filterType={props.walletTransfer.filterPaymentMethodType}
+                        selectedMethodID={props.walletTransfer.selectedAccountID}
+                        shouldShowAddPaymentMethodButton={false}
+                    />
+                </View>
+                <MenuItem
+                    onPress={navigateToAddPaymentMethodPage}
+                    title={props.walletTransfer.filterPaymentMethodType === CONST.PAYMENT_METHODS.BANK_ACCOUNT
+                        ? props.translate('paymentMethodList.addNewBankAccount')
+                        : props.translate('paymentMethodList.addNewDebitCard')}
+                    icon={Expensicons.Plus}
+                    disabled={props.isLoadingPaymentMethods}
```

Comment:
What triggers the payment methods to load? Are we able to even access this page if the payment methods are still loading? Should we be able to access it if they are loading?